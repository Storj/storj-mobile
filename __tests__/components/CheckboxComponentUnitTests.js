import 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import Checkbox from '../../src/components/CheckboxComponent';

Enzyme.configure({ adapter: new Adapter() });


describe('CheckboxComponent', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <Checkbox />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('triggers onPress on TouchableOpacity press', () => {
        const wrapper = shallow(
            <Checkbox 
                onPress={ () => {} } />
        );

        const onPressSpy = sinon.spy(wrapper.instance(), 'onPress');

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(onPressSpy.calledOnce).toBe(true);
    });

    it('triggers setState on TouchableOpacity press', () => {
        const wrapper = shallow(
            <Checkbox 
                onPress={ () => {} } />
        );

        const setStateSpy = sinon.spy(wrapper.instance(), 'setState');

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(setStateSpy.calledOnce).toBe(true);
    });

    it('change state correctly on TouchableOpacity press', () => {
        const wrapper = shallow(
            <Checkbox 
                onPress={ () => {} } 
                isChecked = { false } />
        );

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(wrapper.instance().state.isChecked).toBe(true);
    });

    it('triggers onPress from props on touchableOpacity press', () => {
        const onPressSpy = jest.fn();

        const wrapper = shallow(
            <Checkbox 
                onPress={ onPressSpy } />
        );

        wrapper.find('TouchableOpacity').forEach(child => {
            child.simulate('Press');
        });

        expect(onPressSpy.mock.calls.length).toBe(1);
    });
});