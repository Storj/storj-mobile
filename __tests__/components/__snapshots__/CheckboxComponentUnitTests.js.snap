// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckboxComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Component
        style={null}
      />,
      "onPress": [Function],
      "style": Object {
        "backgroundColor": "transparent",
        "borderColor": "#2684FF",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 48,
        "padding": 4,
        "width": 48,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Component
          style={null}
        />,
        "onPress": [Function],
        "style": Object {
          "backgroundColor": "transparent",
          "borderColor": "#2684FF",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 48,
          "padding": 4,
          "width": 48,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
